### PR TITLE
gh-117804: Autoconf: silence some clang warnings for PGO builds

### PR DESCRIPTION
**Pull request by erlend-aasland**
* `-Wno-profile-instr-unprofiled` for source files without profile data
~~* `-Wno-profile-instr-out-of-date` for _bootstrap_python.c~~ <= solve this in a follow-up PR

* Original issue: gh-117752

* Retargeted to issue: gh-117804

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -2027,7 +2030,10 @@ case "$CC_BASENAME" in
     case $ac_sys_system in
       Darwin*)
```
Comment (erlend-aasland) [inline configure.ac:2031]: BTW (and off-topic for this PR): this check should be for `cc_is_clang`, not `$ac_sys_system = Darwin`

**Comment by vstinner** [comment]
cc @corona10

File: configure.ac (hunk 1 above)
Comment (erlend-aasland) [reply configure.ac:2031]: Will follow up in issue #96398.

**Review by corona10 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Marking this as draft, since I'm not sure we should take this route.

**Comment by erlend-aasland** [comment]
I agree with Steve's comments on the issue. Closing this.

**Pull request closed without merging**